### PR TITLE
bar_5

### PR DESCRIPTION
**Pull request by erinlincoln**
- Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/bar_5.ncl
- Original NCL plots: https://www.ncl.ucar.edu/Applications/Images/bar_5_lg.png
- Issue #340 
image
image

### CODE REVIEW

**Comment by erinlincoln** [comment]
I accidentally made this branch from maponly_2, but I removed that file from this branch so hopefully everything should be fine.

File: Plots/Bar/NCL_bar_5.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+# Show plot
+plt.show()
+
+###############################################################################
+#Although the code above does recreate the NCL plot, it is tedious to replicate
+#it in Python. A different process to use is subplots, which, in the
+#specific example below, each have their own axes. This not only makes the graph
+#easier to read, but is much simpler to recreate. The subplot method is shown
+#below.
+
```
Comment (anissa111) [inline Plots/Bar/NCL_bar_5.py:182]:
```suggestion
###############################################################################
# Although the code above does recreate the NCL plot, it is tedious to replicate
# it in Python. A different process to use is subplots, which, in the
# specific example below, each have their own axes. This not only makes the graph
# easier to read, but is much simpler to recreate. The subplot method is shown
# below.

```

File: Plots/Bar/NCL_bar_5.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+fig2, ax2 = plt.subplots(11, 1, figsize=(12, 24))
+
+# Loop through all of the years and plot them as a bar chart
+i = 0
+for year in warm:
+    # Find year limits of each bar chart
+    year_start = year - 1
+    year_end = year + 3
+
+    # Find the indicies for each of the year limits
```
Comment (anissa111) [inline Plots/Bar/NCL_bar_5.py:196]:
```suggestion
    # Find the indices for each of the year limits
```

File: Plots/Bar/NCL_bar_5.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        tick_list.append("0.0")
+        tick_list.append("1.0")
+        tick_list.append("")
+        i = i + 1
+    tick_list.append("")
+    return tick_list
+
+
+# A helper function to make the right y axis labels
+def right_tick_labels(num_plots, year_list):
```
Comment (anissa111) [inline Plots/Bar/NCL_bar_5.py:117]:
```suggestion
def right_tick_labels(year_list):
```

It doesn't look like you're not using the `num_plots` input to your helper function

File: Plots/Bar/NCL_bar_5.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        i = i + 1
+
+
+# A helper function to create a bar chart for each axis
+def bar_plot(ax_name, year):
+    # Find year limits of each bar chart
+    year_start = year - 1
+    year_end = year + 3
+
+    # Find the indicies for each of the year limits
```
Comment (anissa111) [inline Plots/Bar/NCL_bar_5.py:82]:
```suggestion
    # Find the indices for each of the year limits
```

File: Plots/Bar/NCL_bar_5.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+# Open a netCDF data file using xarray default engine and load the data
+ds = xr.open_dataset(gdf.get("netcdf_files/Jsst.nc"))
+sst = ds.SST
+date = ds.date
+
+#scale the sst variable to range from around -1 to 1
+sst = sst * .1
+
+# Dates in the file are represented by year and month (YYYYMM)
+# representing them fractionally will make ploting the data easier
```
Comment (anissa111) [inline Plots/Bar/NCL_bar_5.py:37]:
```suggestion
# representing them fractionally will make plotting the data easier
```

**Review by anissa111 — CHANGES_REQUESTED** [review]
I like the way you've done both the NCL way and a less tedious way of creating the same plot. 

i've included a few minor comments below, but my main concern is that when I run the code on my local and when I generate the plots, the first plot that is created seems to only show the bottom corner:

image

Also, correct me if I'm wrong, but it seems like you're defining a bunch of helper functions, but then only calling them once, which leads me to believe they don't actually need to be in helper functions.

**Comment by erinlincoln** [comment]
I edited the graphs in my system output rather than the Spyder output, so here is how they now look for me:
bar_5_fig_2
bar_5_fig_1

File: Plots/Bar/NCL_bar_5.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    right_y_ticks.append("")
+    right_y_ticks.append("")
+right_y_ticks.append("")
+
+# Use geocat.viz.util convenience function to set axes limits & tick values without calling several matplotlib functions
+gvutil.set_axes_limits_and_ticks(
+    ax0,
+    xlim=(-3, 3),
+    xticks=np.linspace(-3, 3, 7),
+    xticklabels=["Jan₋₁", "Jul₋₁", "Jan₀", "Jan₀", "Jan₊₁", "Jul₊₁", "Jan₊₂"],
```
Comment (michaelavs) [inline Plots/Bar/NCL_bar_5.py:140]:
```suggestion
    xticklabels=["Jan₋₁", "Jul₋₁", "Jan₀", "Jul₀", "Jan₊₁", "Jul₊₁", "Jan₊₂"],
```

File: Plots/Bar/NCL_bar_5.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
+"""
+NCL_bar_5.py
+===========
```
Comment (michaelavs) [inline Plots/Bar/NCL_bar_5.py:3]:
```suggestion
============
```

File: Plots/Bar/NCL_bar_5.py [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+# Set tick parameters for all axes
+ax0.tick_params(axis="x", length=9, labelsize=12)
+ax0.tick_params(axis="y", length=5, labelsize=8)
+axRHS.tick_params(axis="y", length=6, labelsize=8)
+
+# Show plot
+plt.plot()
+
+###############################################################################
+# Although the code above does recreate the NCL plot, it is tedious to replicate
```
Comment (michaelavs) [inline Plots/Bar/NCL_bar_5.py:159]: @anissa111 based on your comment from maponly, do you think we should have this block as part of its own cell of text, or keep it in this format?

Comment (anissa111) [reply Plots/Bar/NCL_bar_5.py:159]: @michaelavs This is the format I was talking about, yeah! This example correctly formats the comment block for a markdown cell when rendered:

image

**Review by michaelavs — APPROVED** [review]
Looks good to me!

**Review by anissa111 — APPROVED** [review]
Looks good to me, too!

**Pull request merged**